common/src/main/scala/ClassTwoAndThreeResult.scala

```diff
@@ -74,12 +109,13 @@ case class ClassTwoAndThreeResult[A <: ClassTwoOrThree](
     }
 
   def rate: Explained[BigDecimal] = higherRateApplies flatMap {
-    case true => otherRates.at(paymentDate) match {
-      case Some(r) => r.rate gives s"rate: for $paymentDate, due to higher rate"
-      case None =>
-        val mostRecent = otherRates.toList.sortBy(_._1.lowerValue.get.toEpochDay).last
-        mostRecent._2.rate gives s"rate: for ${mostRecent._1}, due to higher rate but no rate defined for $paymentDate"
-    }
+    case true =>
+      val range = Interval.openLower(on, paymentDate)
+      val (when, highest) = otherRates.toList
+        .filter(_._1 intersects range)
+        .sortBy(_._2.rate)
+        .last
```

Comment:
would it not be the rate that would apply on `on` rather than the highest rate available?

Reply:
That's what I thought initially. But it turns out it's the highest. 

Which actually makes sense in retrospect. You wouldn't want people waiting it out if they think they're going to get a cheaper rate later on.